core/src/main/java/org/elasticsearch/plugins/SearchPlugin.java

```diff
@@ -216,4 +222,8 @@ public FetchPhaseConstructionContext(Map<String, Highlighter> highlighters) {
             return highlighters;
         }
     }
+
+    interface SearchResponseListener {
```

Comment:
I just thought of two things for this line:
1. Make sure when you document this you document on which thread pool it is called. It may not be interesting for most readers but I really like to know what thread pool and what node things run on. I guess these run on the search thread pool on the coordinating node, right?
2. We should think about ordering here. One thing we try to do with plugins is make sure that the order that they are imported doesn't matter. I suspect with this the order is the order that we call the listeners. I don't think it'd matter if the classes aren't mutable, but they are. We could fix that by making replacing this interface with something like:
```
static final class SearchResponseListener {
  private final BiFunction<SearchRequest, SearchResponse> listener;
  private final int order;

  ...ctor...

  /** Order in which the listener is executed. It is illegal to register multiple listeners at the same order. */
  public int getOrder() { return order; }
  public void onFinishedSearchResponse(SearchRequest request, SearchResponse response) {
    listener.apply(request, response);
  }
}
```

I dunno if that is the right way to do it, but it is a way.